### PR TITLE
`x` shortcut on sizeless div

### PR DESCRIPTION
**Pull request by bkrmendy**
# Try it [here](https://utopia.pizza/p/256cf461-pie-vermicelli/?branch_name=fix-sizeless-div-x-shortcut)

## Problem
When pressing `x` on a sizeless div, we only add a `width` and `height` prop. We should adding an additional `position: absolute` and appropriate `top, left` props, and the sizeless div's children should have their positions adjusted as well (so that they stay in the same visual position).

## Fix
Implement the above in the shortcut handler.

Additionally, a heuristic rule is added to handle converting into a group (the reverse action of the above). If an element
- is absolutely positioned,
- has more than 0 children,
- all these children are absolutely positioned,
the element is converted into a sizeless div, and the children offsets are adjusted so that they remain visually intact at the same absolute positions.

### CODE REVIEW

File: editor/src/components/editor/shortcuts.spec.browser2.tsx [hunk 1]
```diff
@@ -28,6 +29,8 @@ const StoryBoardId = 'StoryBoardId'
 const ParentId = 'ParentId'
 const backgroundColor = '#384C5CAB'
 
+/* eslint jest/expect-expect: ["error", { "assertFunctionNames": ["expect", "expectBoundsToEqual", "expectNoAction"] }] */
```
Comment (Rheeseyb) [inline editor/src/components/editor/shortcuts.spec.browser2.tsx:32]: Nice!

File: editor/src/components/inspector/inspector-common.ts [hunk 2]
```diff
@@ -516,6 +516,12 @@ export const nukeAllAbsolutePositioningPropsCommands = (
   ]
 }
 
+export const nukeSizingProps = (path: ElementPath): Array<CanvasCommand> => {
```
Comment (Rheeseyb) [inline editor/src/components/inspector/inspector-common.ts:519]: We have `nukeSizingPropsForAxisCommand` which also includes the `min*` and `max*` props, so maybe it's worth either calling that for both axes, or including those props here

File: editor/src/components/editor/global-shortcuts.tsx [hunk 3]
```diff
@@ -903,7 +917,48 @@ export function handleKeyDown(
               return []
             }
 
+            const parentBounds =
```
Comment (Rheeseyb) [inline editor/src/components/editor/global-shortcuts.tsx:920]: I think this entire piece of code (everything under `REMOVE_ABSOLUTE_POSITIONING`) needs to be extracted into its own file plus helper functions

File: editor/src/components/editor/global-shortcuts.tsx [hunk 4]
```diff
@@ -921,6 +976,46 @@ export function handleKeyDown(
...
+              (path) => MetadataUtils.findElementByElementPath(editor.jsxMetadata, path),
+              replaceContentAffectingPathsWithTheirChildrenRecursive(
+                editor.jsxMetadata,
+                editor.allElementProps,
+                MetadataUtils.getChildrenPathsUnordered(editor.jsxMetadata, elementPath),
+              ),
+            )
+
+            return [
+              deleteProperties('always', elementPath, [PP.create('style')]),
```
[the remark this answers is not in the document]
Comment (bkrmendy) [reply editor/src/components/editor/global-shortcuts.tsx]: 😬 fixed in https://github.com/concrete-utopia/utopia/pull/3510/commits/08051de7ece2a962fb4cb23735391b9009e75363

File: editor/src/components/inspector/inspector-common.ts (hunk 2 above)
Comment (bkrmendy) [reply editor/src/components/inspector/inspector-common.ts]: fixed in https://github.com/concrete-utopia/utopia/pull/3510/commits/08051de7ece2a962fb4cb23735391b9009e75363

**Review by Rheeseyb — APPROVED** [review]

**Review by enidemi — APPROVED** [review]

**Pull request merged**